sections/shared/modals/SelectCurrencyModal/SelectCurrencyModal.tsx

```diff
@@ -106,88 +111,196 @@ export const SelectCurrencyModal: FC<SelectCurrencyModalProps> = ({
+						autoFocus={true}
+					/>
+				</SearchContainer>
+				<CategoryFilters>
+					{CATEGORY_FILTERS.map((category) => {
+						const isActive = synthCategory === category;
+						const noItem =
+							synths.filter((synth) => synth.category.toString() === category).length === 0;
 
 						return (
-							<SynthRow
-								key={currencyKey}
+							<CategoryButton
+								variant="secondary"
+								isActive={isActive}
+								disabled={noItem}
 								onClick={() => {
-									onSelect(currencyKey as CurrencyKey);
-									onDismiss();
+									setAssetSearch('');
+									setSynthCategory(isActive ? null : category);
 								}}
-								synthBalance={synthBalances?.balancesMap[currencyKey as CurrencyKey]}
-								{...{ synth }}
-							/>
+								key={category}
+							>
+								{t(`common.currency-category.${category}`)}
+							</CategoryButton>
 						);
-					})
-				) : (
-					<EmptyDisplay>{t('modals.select-currency.search.empty-results')}</EmptyDisplay>
-				)}
-			</RowsContainer>
+					})}
+				</CategoryFilters>
+
+				<InfiniteScroll
+					dataLength={synthsResults.length + oneInchTokensPaged.length}
+					next={() => {
+						setTimeout(() => {
+							setPage(page + 1);
+						}, 200);
+					}}
+					hasMore={oneInchEnabled && oneInchTokensPaged.length !== oneInchTokenList.length}
+					loader={
+						<LoadingMore>
+							<Loader inline />
+						</LoadingMore>
+					}
+					scrollableTarget="scrollableDiv"
+				>
+					<RowsHeader>
+						<span>
+							{assetSearch ? (
+								<span>{t('modals.select-currency.header.search-results')}</span>
+							) : synthCategory != null ? (
+								t('modals.select-currency.header.category-synths', {
+									category: synthCategory,
+								})
+							) : (
+								t('modals.select-currency.header.all-synths')
+							)}
+						</span>
+						<span>{t('modals.select-currency.header.holdings')}</span>
+					</RowsHeader>
+					{synthsWalletBalancesQuery.isLoading ? (
+						<Loader />
+					) : synthsResults.length > 0 ? (
+						// TODO: use `Synth` type from contracts-interface
+						synthsResults.map((synth) => {
+							const currencyKey = synth.name;
+
+							return (
+								<CurrencyRow
+									key={currencyKey}
+									onClick={() => {
+										onSelect(currencyKey, false);
+										onDismiss();
+									}}
+									balance={synthBalances?.balancesMap[currencyKey as CurrencyKey]}
+									token={{
+										name: synth.description,
+										symbol: synth.name,
+										isSynth: true,
+									}}
+								/>
+							);
+						})
+					) : (
+						<EmptyDisplay>{t('modals.select-currency.search.empty-results')}</EmptyDisplay>
+					)}
+					{oneInchTokensPaged.length ? (
+						<>
+							<TokensHeader>
+								<span>
+									{assetSearch ? (
+										<span>{t('modals.select-currency.header.search-results')}</span>
+									) : synthCategory != null ? (
+										t('modals.select-currency.header.category-synths', {
```

Comment:
This will display "Crypto Synths" or "Forex Synths" above the 1inch tokens when there is a filter selected

<img width="422" alt="image" src="https://user-images.githubusercontent.com/10401554/176785531-26960898-100e-4658-81f0-255191af2c8c.png">

Reply:
Good spot, I think I'll hide 1inch tokens when forex is selected and then update the label for when crypto is selected